### ISSUE TITLE
Metric names when using multiple dataloader for validation

### ISSUE THREAD

**Issue by kantholtz** [PyTorchLightning/pytorch-lightning#4848]
Hi guys,

I am using multiple dataloaders for validation. This works great so far, but I have some questions regarding the logged metrics:

As far as I understand, lightning will automatically assign a metric suffix (`/dataloader_idx_X`). Is there any way for me to control that behaviour? Wandb assumes groups with that slash, and it means, that it groups by metric name and not validation set. However, there are scenarios where it makes much more sense to group by validation set, as they might look at completely different metrics. On the bottom, there is a screenshot on how it looks without any further wanbd configuration.

Apart from my question, this is more of a feature proposal: Can't we supply a (ordered) dictionary instead of a list via `val_dataloader()` in DataModules? So that it looks like this:

DataModule:
```python
def val_dataloader(self) -> Union[Dict[str, DataLoader], List[Dataloader], DataLoader]:
   return {
      'scenario1': DataLoader('...'),
      'scenario2': DataLoader('...'),
  }
```

LightningModule
```python
def validation_step(self, batch, batch_idx: int, loader: Union[str, int]):
   # do something
   self.log('my_metric', x)
```
Which then logs `scenarioX/my_metric` in case of the example or `dataloader_idx_X/my_metric` when a list is provided.

![Screenshot_20201125_095000](https://user-images.githubusercontent.com/608779/100204025-a257c300-2f03-11eb-9d79-47073d3aa2a9.png)

**Comment by tchaton** [comment]
Hey @kantholtz,

Thanks for sharing your ideas ! It shouldn't be too hard to add.
Let me see if I have some time for it this week.

Best regards,
T.C

**Comment by tulay** [comment]
@kantholtz would you please mention PL version you use? When using multiple validation loaders, I am having issues with 1.0.8 most likely due to the[ line ](https://github.com/PyTorchLightning/pytorch-lightning/blob/1.0.8/pytorch_lightning/trainer/connectors/logger_connector.py#L229).

**Comment by kantholtz** [comment]
Sure, I am using `1.1.0-dev`

**Comment by tulay** [comment]
Thanks. I pulled "pytorch-lightning==1.1.0rc1" and it runs fine.

**Comment by kantholtz** [comment]
@tchaton quick question: is there a single place, where I can simply remove the suffix without too much hassle or is it integrated more deeply? (My metrics have distinct names so there won't be any collision.)

**Comment by kantholtz** [comment]
For the curious if anyone needs a quick and dirty workaround: I set https://github.com/PyTorchLightning/pytorch-lightning/blob/7b7d4bb86d2c28ccc8b13953481f084d6f8accf7/pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py#L152 to `False` and the suffix vanishes. 

Quick and dirty ;)
